src/algebra/gcd_monoid/integrally_closed.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebra.gcd_monoid.basic
+import ring_theory.integrally_closed
+import ring_theory.polynomial.scale_roots
+import ring_theory.polynomial.eisenstein
+
+/-!
+
+# GCD domains are integrally closed
+
+-/
+
+open_locale big_operators polynomial
+
+lemma is_localization.surj_of_gcd_domain {R : Type*} [comm_ring R] [is_domain R] [gcd_monoid R]
+  (M : submonoid R) {K} [comm_semiring K] [algebra R K] [is_localization M K] (z : K) :
+  ∃ a b : R, is_unit (gcd a b) ∧ z * algebra_map R K b = algebra_map R K a :=
+begin
+  obtain ⟨x, ⟨y, hy⟩, rfl⟩ := is_localization.mk'_surjective M z,
+  obtain ⟨x', y', d, rfl, rfl, hu⟩ := extract_gcd x y,
+  use [x', y', hu],
+  rw [mul_comm, is_localization.mul_mk'_eq_mk'_of_mul],
+  convert is_localization.mk'_mul_cancel_left _ _ using 2,
+  { rw [← mul_assoc, mul_comm y'], refl }, { apply_instance },
+end
+
+lemma dvd_pow_nat_degree_of_eval₂_eq_zero {R A : Type*} [comm_ring R] [comm_ring A] {f : R →+* A}
```

Comment:
It can go right after `_root_.scale_roots.is_weakly_eisenstein_at`.

Reply:
It can even go to `ring_theory.polynomial.scale_roots`. You cannot use `is_weakly_eisenstein` there, but you can probably just prove it by hand (`ring_theory.polynomial.eisenstein` has quite a lot of imports).